File: Leetcode/3105. Longest Strictly Increasing or Strictly Decreasing Subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        n=len(nums)
        t1,t2=nums[0],nums[0]
        res1=1
        res2=1
        res=1
        for i in range(1,n):
            if t1>nums[i]:
                res1+=1
                res=max(res,res1)
            else:
                res1=1
            if t2<nums[i]:
                res2+=1
                res=max(res,res2)
            else:
                res2=1
            t1,t2=nums[i],nums[i]
        return res

File: Leetcode/3105. Longest Strictly Increasing or Strictly Decreasing Subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int n=nums.size();
        int t1=nums[0],t2=nums[0];
        int res1=1,res2=1,res=1;
        for(int i=1;i<n;i++){
            if(t1>nums[i]){
                res=max(++res1,res);
            }else{
                res1=1;
            }
            if(t2<nums[i]){
                res=max(res,++res2);
            }else{
                res2=1;
            }
            t1=nums[i];
            t2=nums[i];
        }
        return res;
    }
};
